### PR TITLE
Change migration name to avoid conflicts

### PR DESCRIPTION
**Pull request by arpitchauhan**
The migration AddMissingIndexes conflicts with a migration of similar
name in other gems, eg acts_as_taggable_on 4.0.0. Its name has been
made more specific in order to avoid such conflicts while installing
migrations in a rails app using

```
bundle exec rake railties:install:migrations
```

### CODE REVIEW

**Comment by krzysiek1507** [comment]
Hi @arpitchauhan could you please rebase branch and create PR for `master` too?

**Comment by arpitchauhan** [comment]
@krzysiek1507 Rebasing is proving quite complicated. Is it ok if I cherry-pick my commit onto a new branch made out of master?

**Comment by krzysiek1507** [comment]
@arpitchauhan sorry. I mean, rebase your branch with `3-1-stable` and also create new PR to `master` with cherry-picked commits. ;) I hope now it's clear.

**Comment by arpitchauhan** [comment]
Sure, will do.

**Comment by arpitchauhan** [comment]
@krzysiek1507 Done. The other PR is #7530

**Comment by krzysiek1507** [comment]
Hi @arpitchauhan could you add checking if indexes exist?

**Comment by arpitchauhan** [comment]
@krzysiek1507 do you mean adding spec examples (tests)?

**Comment by arpitchauhan** [comment]
@krzysiek1507 Ah, got you. Will add them.

**Comment by arpitchauhan** [comment]
@krzysiek1507 done

**Comment by krzysiek1507** [comment]
@arpitchauhan could you squash commits please?

**Comment by arpitchauhan** [comment]
@krzysiek1507 done

File: core/db/migrate/20150314013438_add_missing_indexes_on_spree_tables.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+class AddMissingIndexesOnSpreeTables < ActiveRecord::Migration
+  def change
+    unless index_exists?(:spree_promotion_rules_users, [:user_id, :promotion_rule_id])
+      add_index   :spree_promotion_rules_users,
```
Comment (krzysiek1507) [inline core/db/migrate/20150314013438_add_missing_indexes_on_spree_tables.rb]: Shouldn't we consider index name in `index_exists?`? And please fix indentation.

Comment (arpitchauhan) [inline core/db/migrate/20150314013438_add_missing_indexes_on_spree_tables.rb]: @krzysiek1507 The [documentation](http://apidock.com/rails/ActiveRecord/ConnectionAdapters/SchemaStatements/index_exists%3F) says we need to use table name and column name(s) as arguments.

I'll fix the spacing between `add_index` and arguments.

Comment (arpitchauhan) [inline core/db/migrate/20150314013438_add_missing_indexes_on_spree_tables.rb]: @krzysiek1507  done

Comment (krzysiek1507) [inline core/db/migrate/20150314013438_add_missing_indexes_on_spree_tables.rb]: @arpitchauhan Yes, but we can put `name` into `options` hash (4th example).
If you check source, you will see that name will be consider too.

Comment (arpitchauhan) [inline core/db/migrate/20150314013438_add_missing_indexes_on_spree_tables.rb]: @krzysiek1507 got it, thanks. Done.

**Comment by arpitchauhan** [comment]
@krzysiek1507 is it possible for this PR to be included in 3.1.2?

**Comment by arpitchauhan** [comment]
@damianlegawiec Similar MR for master branch was merged ( #7530 ). Can this be merged too?

**Comment by damianlegawiec** [comment]
Sure @arpitchauhan  and thanks for the fix!

**Pull request merged**